dbt/include/dremio/macros/materializations/twin_strategy.sql

```diff
@@ -14,25 +14,45 @@ limitations under the License.*/
 
 {%- macro apply_twin_strategy(target_relation) -%}
   {%- set twin_strategy = config.get('twin_strategy', validator=validation.any[basestring]) or 'clone' -%}
+  
+  {%- set view_relation = api.Relation.create(
+    identifier=generate_alias_name_impl(model.name, config.get('alias', validator=validation.any[basestring]), model),
+    schema=generate_schema_name_impl(target.schema, config.get('schema', validator=validation.any[basestring]), model),
+    database=generate_database_name_impl(target.database, config.get('database', validator=validation.any[basestring]), model),
+    type='view') -%}
+
   {%- if target_relation.type == 'view' -%}
-    {%- if twin_strategy != 'allow' -%}
-      {%- set table_relation = api.Relation.create(
-          identifier=generate_alias_name_impl(model.name, config.get('file', validator=validation.any[basestring]), model),
-          schema=generate_schema_name_impl(target.root_path, config.get('root_path', validator=validation.any[basestring]), model),
-          database=generate_database_name_impl(target.datalake, config.get('datalake', validator=validation.any[basestring]), model),
-          type='table') -%}
-      {{ adapter.drop_relation(table_relation) }}
+    -- Check if there is a conflicting table
+    {%- set table_relation = api.Relation.create(
+        identifier=generate_alias_name_impl(model.name, config.get('file', validator=validation.any[basestring]), model),
+        schema=generate_schema_name_impl(target.root_path, config.get('root_path', validator=validation.any[basestring]), model),
+        database=generate_database_name_impl(target.datalake, config.get('datalake', validator=validation.any[basestring]), model),
+        type='table') -%}
+    {%- set conflicting_table = adapter.get_relation(database=table_relation.database, schema=table_relation.schema, identifier=table_relation.identifier) -%}
+    {%- if conflicting_table is not none -%}

```

Comment:
Could we reduce if\else statements and use only 1 level of integrity?

Reply:
Done